source/gui/accPropServer.py

```diff
@@ -90,24 +91,53 @@ def _getPropValue(self, pIDString, dwIDStringLen, idProp):
 			to extract the HWND/idObject/idChild from the identity string.
 			Note that, while one IAccPropServer implementation can annotate
 			multiple accessible elements, it is still bound to one wx.Control.
-		@type pIDString: str
 		@param dwIDStringLen: Specifies the length of the identity string specified by the pIDString parameter.
-		@type dwIDStringLen: int
-		@param idProp: Specifies a GUID indicating the desired property.
-		@type idProp: One of the oleacc.PROPID_* GUIDS
-		@return A tuple of the out params for the `IAccPropServer::GetPropValue` method: `VARIANT* pvarValue` and `BOOL*
-		pfHasProp`. When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue part must be VT_EMPTY.
-		Consider using self.NO_RETURN_VALUE instead. Returning (VT_EMPTY, HAS_PROP) IS valid, meaning the property exists
-		but is empty.
+		@param idProp: Specifies a GUID indicating the desired property. One of the values from oleacc.PROPID_*
+		@return Use L{self._hasProp} to return correct values or return None if unable to supply the property.
 		"""
 		raise NotImplementedError
 
-	def GetPropValue(self, pIDString, dwIDStringLen, idProp):
+	def _hasProp(self, value: Any) -> Optional[Tuple[BOOL, Any]]:
+		"""Constructs a tuple for the `IAccPropServer::GetPropValue` method, two elements:
+			1. `VARIANT pvarValue`
+			2. `BOOL pfHasProp` (either self.HAS_PROP or self.DOES_NOT_HAVE_PROP)"""
+		return value, self.HAS_PROP
+
+	def GetPropValue(
+			self, this, # unused "this" used to indicate to comTypes we want a low level implementation
+			pIDString: str,
+			dwIDStringLen: int,
+			idProp: GUID,
+			pvarValue,
+			pfGotProp
+	) -> int:
+		""" Exposed method to get a prop value.
+		see L{_getPropValue} for more details of args.
+		Uses a low-level approach, because comtypes tries to clear the VARIANT even though it is an out param.
+		When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue.vt part must be VT_EMPTY.
+		@type pvarValue: POINTER[VARIANT]
+		@type pfGotProp: POINTER[BOOL]
+		"""
+		# Set values in case we return early.
+		pfGotProp[0] = self.DOES_NOT_HAVE_PROP
+		pvarValue[0].vt = VT_EMPTY
+		try:
+			ret = self._getPropValue(pIDString, dwIDStringLen, idProp)
+			if ret is None:
+				return S_OK
+			elif len(ret) != 2:
+				raise RuntimeError("_getPropValue implementation must return None or two element tuple")
```

Comment:
I'm afraid the caller really won't like this. In this case the server and the client are the same, but I think it is better to log an error here and return a hresult that indicates an error.

Reply:
These are both caught and logged below, perhaps this makes the control flow unclear. I'll see what I can do to make it clearer. I can't think of a situation where it is beneficial to notify the caller of an internal error, especially given the caller is something in IAccessible. . It certainly helps to log these exceptions though, otherwise they don't make it to the log.

Reply:
Ah I'm sorry, I see it now. I wasn't keeping an eye on the indentation properly.
I guess it is ok if we return something else than S_OK on errors.